### ISSUE TITLE
warning: unknown lint: `crown_is_not_used`

### ISSUE THREAD

**Issue by Loirooriol** [servo/servo#32597]
Probably due to #32494, when compiling with `./mach build -r`, I get

```
warning: unknown lint: `crown_is_not_used`
  --> components/script/lib.rs:13:9
   |
13 | #![deny(crown_is_not_used)]
   |         ^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unknown_lints)]` on by default
```

@jschwe

**Comment by wusyong** [comment]
Yeah it's a warning if `--use-crown` isn't enabled and cargo saw an unknown lint.

**Comment by jschwe** [comment]
During review of #32494 it was explicitly requested [to keep the warning enabled](https://github.com/servo/servo/pull/32494#discussion_r1639773126). 
CC @sagudev

I think the idea here is to take a conservative approach, and first be very sure that the `--use-crown` feature is working as expected, before later relaxing things and silencing / removing the warning.

**Comment by sagudev** [comment]
Like jschwe said, that warning is intended, but we could rename it to make it sound less scary.
